Update "Strategy Explained" and "Results Explained" (#67)
dema-trading-ai/docs: docs/your_first_strategy/results_explained.md

Once you have installed the Engine following the instructions in
[Section 1.1: Installation](/installation_and_updating/installation), simply navigate to the folder where you created
your project, and run the command ‘engine’. This will run the Engine, and you will be presented with the results of
your strategy. The results are divided into a few categories. The first one is: the Main Results.

## Main Results
![Main Results](imgs/main_results.png) <br>
This is how the first section of your results look like. The Main Results are divided into three separate tables:
Settings, Performance, and Trade Info.

### Settings
This table shows the settings that you used while testing your algorithm. If you are unsure what the different
settings mean, we explain them in
[Section 2.5: Editing Your Configuration](/your_first_strategy/editing_your_configuration).

### Performance
This table shows the performance of the strategy (on portfolio level, so not divided by coin pair). A brief
explanation of each of the terms used:

* **End capital**: total amount of capital at the end of backtesting period. This capital is the sum of the budget

> (that can be used to start new trades) and the total worth of your open trades (converted to your base currency).

Author: How about:
total amount of capital at the end of backtesting period. This capital is the sum of your current budget (the available funds that can be used to start new trades) and the total worth of your open trades (converted to your base currency).

> (that can be used to start new trades) and the total worth of your open trades (converted to your base currency).

Author: I added this in a new commit - let me know if you agree or if you think it should be different still.

* **Max. seen drawdown**: for a more detailed explanation, see the [Termsheet](/knowledge_library/term_sheet).
* **Max. seen drawdown from**: the starting date of the drawdown period where the max. seen drawdown occurred.
* **Max. seen drawdown to**: the end date of the drawdown period where the max. seen drawdown occurred. If this is a
‘-’ instead of a date, that means that the drawdown period has not yet ended when the backtest concluded.
* **Max. seen drawdown at**: the specific date and time at which the max. seen drawdown occurred.
* **Market change coins**: the average market change of the whitelisted trading pairs within the backtesting period.
* **Market change BTC**: the market change of the pair BTC/USDT within the backtesting period.
* **Market drawdown coins**: max. seen drawdown of the market change of the whitelisted trading pairs.
* **Market drawdown BTC**: max. seen drawdown of the market change of the pair BTC/USDT.
* **Total fee paid**: summed total of all paid fees (including open trade fees)

### Trade Info
This table shows the trade information combined over all whitelisted pairs. A brief explanation of each of the
terms used:

> * **Amount of trades**: total number of trades the strategy has been able to open (buy) and close (sell).

Author: How about:
total number of trades the strategy has been able to open (buy) and close (sell). Left-open trades (trades that are bought but not sold yet, by the time the backtesting period ended) are not counted in this.

> * **Amount of trades**: total number of trades the strategy has been able to open (buy) and close (sell).

Author: I added this in a new commit - let me know if you agree or if you think it should be different still.

* **Max. seen drawdown**: for a more detailed explanation, see the [Termsheet](/knowledge_library/term_sheet/#seen-drawdown).
* **Max. seen drawdown from**: the starting date of the drawdown period where the max. seen drawdown occurred.
* **Max. seen drawdown to**: the end date of the drawdown period where the max. seen drawdown occurred. If this is a
‘-’ instead of a date, that means that the drawdown period has not yet ended when the backtest concluded.
* **Max. seen drawdown at**: the specific date and time at which the max. seen drawdown occurred.
* **Market change coins**: the average market change of the whitelisted trading pairs within the backtesting period.
* **Market change BTC**: the market change of the pair BTC/USDT within the backtesting period.
* **Market drawdown coins**: max. seen drawdown of the market change of the whitelisted trading pairs.
* **Market drawdown BTC**: max. seen drawdown of the market change of the pair BTC/USDT.
* **Total fee paid**: summed total of all paid fees (including open trade fees)

### Trade Info
This table shows the trade information combined over all whitelisted pairs. A brief explanation of each of the
terms used:

> * **Amount of trades**: total number of trades the strategy has been able to open (buy) and close (sell). Left-open

Reviewer: ```suggestion
* **Amount of trades**: total number of times the strategy has opened (buy) and closed (sell) a trade. Left-open
```

Underneath the Main Results section is the Coin Insights section. This section shows the results of the trading
strategy, separated per whitelisted coin. This can give you a good insight into how your strategy performance
differs between coins.

### Coin Performance
The first table in the Coin Insights section is the Coin Performance table. This shows the performance of the
strategy on coin level. A brief description of the terms used:

* **Avg. profit (%)**: the average profit percentage over all trades for that coin.
* **Cum. profit (%)**: the cumulative profit percentage over all trades for that coin. Here, we define cumulative
as the sum over all trade profit percentages.
* **Total profit (%)**: Total profit percentage over all trades for that coin. This is defined as the product over
all trade profit percentages.
* **Profit ($)**: Total amount of profit in USDT over all trades for that coin. For calculating the profit (in USDT)
for a particular pair, the fee for both opening and closing a trade is subtracted from the profit.

### Coin Metrics
The second table in the Coin Insights section is the Coin Metrics table. This table shows important metrics for
each pair. 

> * **Market change (%)**: the market change for the corresponding pair.

Reviewer: In this section you often use 'pair' while above you use coin. I think it is better to decide on one and be consistent with it.

Author: We use it interchangeably in our code too. I'll change all references to 'coin' since that's what we use most in our result overview

* **Opened at**: the date and time at which the trade has been opened.

## Plots
If you want a more detailed view of how your algorithm performed over time, you can find two categories of plots
that can help you achieve that. After you run the Engine for the first time, the plots will be located in the folder
DIRECTORY_NAME/data/backtesting-data/plots.

### Coin Plot
![Coin Plot](imgs/coin_plot.png) <br>
The coin plots are located directly in the plots folder, and are named plotPAIRNAME.html (for example the BTC/USDT
Plot is in the file plotBTCUSDT.html). The plot shows the fluctuation of the value of the coin. If you zoom in, you
can see the OHLC candle for each timestep. The bottom graph by default shows the volume of the coin at each timestep.
You can add or change any indicators shown in these plots, by changing your config file. Find out how to do that in
[Section 2.5: Editing Your Configuration](/your_first_strategy/editing_your_configuration).

### Equity Plot
![Equity Plot](imgs/equity_plot.png) <br>
The equity plot can be found in the plots/equity folder. This graph shows the capital of your trading algorithm over

> time. The horizontal parts of the graph signify moments where the algorithm had no trades open.

Reviewer: ```suggestion
time. The horizontal parts of the graph signify moments where the algorithm had no trades open.

Depending on your preference, you can toggle between a linear view and a logarithmic view of the y-axis. Furthermore, in the top right of the plot you will find a menubar with several functionalities like: export, zoom, pan, and more.
```

}
```

This includes the following statistics:

* **Status**: Open or Closed - signifies the current status of the trade.
* **Opened at**: The date and time the trade was opened.
* **Closed at**: The date and time the trade was closed. Will be null when the trade is still open.
* **Pair**: The pair that was traded.
* **Open price**: The price in base currency on the moment the trade was opened
* **Close price**: The price in base currency on the moment the trade was closed. Will be null when the trade is still
open.
* **Fees paid**: The amount that was paid in buy- and sell-fees.
* **Starting amount**: The amount of base currency that the trade was worth when it was opened.
* **Capital**: The amount of base currency that the trade was worth when it was sold, or the current amount that it
was worth at the end of the backtesting period, if trade is still open.
* **Currency amount**: The amount of the Crypto currency that was bought in the trade.
* **Sell reason**: The reason why the trade was closed. Can be one of the following: SELL_SIGNAL (when it was closed
because a sell signal was sent), ROI (when it was closed because one of the ROI targets was hit), STOPLOSS (when it

> was closed because the stoploss was hit), or NONE (when the trade is still open).

Reviewer: Do you already want to update this with the new fee attributes?

Author: Done!
